swagger.yaml

```diff
@@ -460,6 +460,11 @@ paths:
           required: false
           type: number
           description: Use together with `lat` and `long`. Returns only resources that are within the given `distance` of the point which is calculated from `lat` and `lon`.
+        - name: free_of_charge
+          in: query
+          required: false
+          type: boolean
+          description: If given boolean is `true`, returns only resources that have their `min_price_per_hour` value of `0` or `None`. If given boolean is `false`, returns all resources.  
```

Comment:
Maybe this was already discussed, but shouldn't the logical behavior be to return _only paid_ resources if `free_of_charge` is set to `false`? If the API user doesn't want to filter based on the charge status, they can leave out the filter altogether.

Reply:
@juyrjola as far as I know, by default respa sends the parameter as `false` on all requests when fetching the resources, so if we implement the filter as suggested, Varaamo needs some refactoring. Another solution could be to rename the filter to `only_free_of_charge`, as suggested by @vikoivun (I think?) on the original issue #407, so it would make more sense.

Reply:
Currently Varaamo (as deployed at varaamo.hel.fi) appears to use a query param called `charge`, which is unsupported anyway. Looking at the Varaamo code, changing it to only send the query param if the filter is active looked like a pretty small change (it can be done e.g. here: https://github.com/City-of-Helsinki/varaamo/blob/feature/helsinki-brand-v2/app/utils/searchUtils.js#L8)

Reply:
So something like this, and a small change in Varaamo → voilà!
 
```python
    def filter_free_of_charge(self, queryset, name, value):
        qs = Q(min_price_per_hour__lte=0) | Q(min_price_per_hour__isnull=True)
        if value:
            return queryset.filter(qs)
        else:
            return queryset.exclude(qs)
```